### PR TITLE
change logging stuff

### PR DESCRIPTION
**Pull request by brotzeit**
I think `special-mode` is better than `view-mode`

### CODE REVIEW

File: lsp-mode.el [hunk 1]
```diff
@@ -5592,36 +5587,58 @@ SESSION is the active session."
...
+      (with-current-buffer (lsp--get-log-buffer-create w)
+        (erase-buffer)))))
+
+(defun lsp--log-io-next (arg)
+  "Move to next log entry."
   (interactive "P")
   (ewoc-goto-next lsp--log-io-ewoc (or arg 1)))
 
-(defun lsp-log-io-prev (arg)
+(defun lsp--log-io-prev (arg)
```
Comment (yyoncho) [inline lsp-mode.el]: Why do you rename the function? We previously decided to make this public so a user could eventually re-map them.

File: lsp-mode.el [hunk 2]
```diff
@@ -5592,36 +5587,58 @@ SESSION is the active session."
...
+(defun lsp--get-log-buffer-create (workspace)
+  "Return the lsp log buffer of WORKSPACE, creating a new one if needed."
+  (let ((server-id (-> workspace lsp--workspace-client lsp--client-server-id symbol-name))
+        (pid (format "%s" (process-id (lsp--workspace-cmd-proc workspace)))))
+    (get-buffer-create (format "*lsp-log: %s:%s*" server-id pid))))
+
+(defun lsp--erase-log-buffer ()
+  "Delete contents of the project's lsp log buffers."
+  (interactive)
+  (let* ((session (lsp-session))
```
Comment (yyoncho) [inline lsp-mode.el]: session is not used.

File: lsp-mode.el [hunk 3]
```diff
@@ -5592,36 +5587,58 @@ SESSION is the active session."
...
 
-(defun lsp-log-io-next (arg)
+(defun lsp--get-log-buffer-create (workspace)
+  "Return the lsp log buffer of WORKSPACE, creating a new one if needed."
+  (let ((server-id (-> workspace lsp--workspace-client lsp--client-server-id symbol-name))
+        (pid (format "%s" (process-id (lsp--workspace-cmd-proc workspace)))))
+    (get-buffer-create (format "*lsp-log: %s:%s*" server-id pid))))
+
+(defun lsp--erase-log-buffer ()
+  "Delete contents of the project's lsp log buffers."
```
Comment (yyoncho) [inline lsp-mode.el]: project -> session.

Binding this to `k` feels a bit unnatural to me. What do you think about having `k` to delete only the current buffer logs and `K` to delete all?

File: lsp-clients.el [hunk 4]
```diff
@@ -27,6 +27,7 @@
 (require 'lsp)
 (require 'dash)
 (require 'dash-functional)
+(require 'rx)
```
Comment (MaskRay) [inline lsp-clients.el]: rx is not used?

File: lsp-mode.el [hunk 5]
```diff
@@ -5592,36 +5587,58 @@ SESSION is the active session."
...
     (lsp--spinner-stop)))
 
+;; lsp-log-io-mode
+
+(defvar lsp-log-io-mode-map
+  (let ((map (make-sparse-keymap)))
+    (define-key map (kbd "M-n") #'lsp--log-io-next)
+    (define-key map (kbd "M-p") #'lsp--log-io-prev)
+    (define-key map (kbd "k") #'lsp--erase-log-buffer)
+    map)
```
Comment (seagle0128) [inline lsp-mode.el]: The key bindings seem not conventional for Emacs.

File: lsp-mode.el (hunk 1 above)
Comment (seagle0128) [reply lsp-mode.el]: Agree. Don't change the public APIs.

File: lsp-clients.el (hunk 4 above)
Comment (brotzeit) [reply lsp-clients.el]:
```elisp
(defun lsp-typescript-javascript-tsx-jsx-activate-p (filename &optional _)
  "Check if the javascript-typescript language server should be enabled based on FILENAME."
  (or (string-match-p (rx (one-or-more anything) "." (or "ts" "js") (opt "x") string-end) filename)
      (derived-mode-p 'js-mode 'js2-mode 'typescript-mode)))
```

File: lsp-mode.el (hunk 5 above)
Comment (brotzeit) [reply lsp-mode.el]: I don't use regular emacs bindings. What would you suggest ?

Comment (seagle0128) [reply lsp-mode.el]: I would suggest `M-k` to erase the whole buffer.
Or as @yyoncho's suggestion:

>Binding this to k feels a bit unnatural to me. What do you think about having k to delete only the current buffer logs and K to delete all?

File: lsp-mode.el [hunk 6]
```diff
@@ -5592,36 +5587,67 @@ SESSION is the active session."
       (lsp--start-workspace session client project-root (lsp--create-initialization-options session client))
     (lsp--spinner-stop)))
 
+;; lsp-log-io-mode
+
+(defvar lsp-log-io-mode-map
+  (let ((map (make-sparse-keymap)))
+    (define-key map (kbd "M-n") #'lsp--log-io-next)
```
Comment (yyoncho) [inline lsp-mode.el]: you have forgotten to update the function names.

**Pull request merged**